docs/admin/prometheus.md

```diff
@@ -0,0 +1,50 @@
+# Prometheus
+
+Coder has support for Prometheus metrics using the dedicated [Go client library](https://github.com/prometheus/client_golang). The library exposes various [metrics types](https://prometheus.io/docs/concepts/metric_types/), such as gauges, histograms, and timers, that give insight into the live Coder deployment.
+
+Feel free to browse through the [Getting started](https://prometheus.io/docs/prometheus/latest/getting_started/) guide, if you don't have an installation of the Prometheus server.
+
+## Enable Prometheus metrics
+
+Coder server exports metrics via the HTTP endpoint, which can be enabled using either the environment variable `CODER_PROMETHEUS_ENABLE` or the flag` --prometheus-enable`.
+
+Use either the environment variable `CODER_PROMETHEUS_ADDRESS` or the flag ` --prometheus-address <network-interface>:<port>` to select a custom endpoint.
+
+If `coder server --prometheus-enable` is started locally, you can preview the metrics endpoint: <!-- markdown-link-check-disable -->http://localhost:2112/<!-- markdown-link-check-enable --> (default endpoint).
+
+```
+# HELP coderd_api_active_users_duration_hour The number of users that have been active within the last hour.
+# TYPE coderd_api_active_users_duration_hour gauge
+coderd_api_active_users_duration_hour 0
+# HELP coderd_api_concurrent_requests The number of concurrent API requests
+# TYPE coderd_api_concurrent_requests gauge
+coderd_api_concurrent_requests 2
+# HELP coderd_api_concurrent_websockets The total number of concurrent API websockets
+# TYPE coderd_api_concurrent_websockets gauge
+coderd_api_concurrent_websockets 1
+# HELP coderd_api_request_latencies_ms Latency distribution of requests in milliseconds
+# TYPE coderd_api_request_latencies_ms histogram
+coderd_api_request_latencies_ms_bucket{method="GET",path="",le="1"} 10
+coderd_api_request_latencies_ms_bucket{method="GET",path="",le="5"} 13
+coderd_api_request_latencies_ms_bucket{method="GET",path="",le="10"} 14
+coderd_api_request_latencies_ms_bucket{method="GET",path="",le="25"} 15
+...
+```
+
+## Available collectors
```

Comment:
This section seems confusing, I think we should omit it entirely. People can discover what metrics we have by querying their prometheus server i.e. through grafana, which should show all of the comments for every metric.

Reply:
https://github.com/coder/coder/issues/3520#issuecomment-1240970387

I agree with that statement as it strictly depends on the customers. Some users prefer to read about available metrics in docs, but the rest would simply hit the endpoint using the browser. Personally, I would keep this section, but I'm also fine with removing it.

Reply:
I think we should document this too... I'd be hesitant to set up prometheus/grafana with Coder if I didn't know what metrics I got. Gitlab does this well IMO (I suspect it's auto-generated): https://docs.gitlab.com/ee/administration/monitoring/prometheus/gitlab_metrics.html#metrics-available

I agree the format is a bit confusing compared to gitlab's

Reply:
Theoretically, we could keep all/most metrics in a single registry/structure, and then just generate a listing for docs purposes. Frankly speaking, a generator seems to be the only reasonable method to keep the docs up-to-date as manual updates in docs will become sooner or later outdated.

Does it sound like a separate issue or do we want to introduce it here?

Reply:
I suppose it can be a separate issue but it would be nice if this doc represented the current metrics / desired structure of the autogenerated stuff.

Reply:
Ok, I implemented a basic generator that uses the provided (and versioned) Prometheus dump. Let me know what you think.

See [preview](https://github.com/mtojek/coder/blob/3520-doc-prom-metrics/docs/admin/prometheus.md#available-metrics)

Reply:
This is great!!